miner/src/miner.rs

```diff
@@ -393,18 +394,41 @@ impl MinerService for Miner {
 	}
 
 	fn pending_transactions_hashes(&self) -> Vec<H256> {
-		let transaction_queue = self.transaction_queue.lock().unwrap();
-		transaction_queue.pending_hashes()
+		match (self.sealing_enabled.load(atomic::Ordering::Relaxed), self.sealing_work.lock().unwrap().peek_last_ref()) {
```

Comment:
if we are maintaining a pending block, then we return the transactions within that block. if we are not maintaining a pending block, then we replicate the old behaviour of just using all submitted (in queue) transactions.

really, there should be another tag aside from `pending`, something like `submitted`, which does the original behaviour, and `pending` should mean only those transactions which have been executed locally (and have according receipts).